state_machine/drivers/pycubedmini/lib/pycubed.py

```diff
@@ -478,19 +467,13 @@ def timeon(self):
         """ return the time on a monotonic clock """
         return int(time.monotonic()) - self.BOOTTIME
 
-    def zero_flags(self):
-        """ zero all flags in non volatile memory """
-        self.f_contact = 0
-        self.f_burn = 0
-
-    def zero_counters(self):
-        """ zero all counters in non volatile memory """
-        self.c_boot = 0
-        self.c_state_err = 0
-        self.c_vbus_rst = 0
-        self.c_deploy = 0
-        self.c_downlink = 0
-        self.c_logfail = 0
+    def clear_nvm(self):
```

Comment:
Have you tried this? I'm not sure how the nvm addressing works. Is there circuitpython or program files stored in nvm that this could mess up?

Reply:
It works, and no it shouldn't mess anything up.
I tried it on the mainboard I have with no issues. 

My uncle actually by coincidence explained to me how NVM works for circuitpython a couple days ago, they just partition a region of flash space for NVM.

Reply:
Also if you look at the bitFlags class, you see that it just modifies NVM the same way.
```python3
class bitFlag:
    '''
    Single bit register WITHIN a byte that can be read or set
    values are 'bool'

    Assumes register is read MSB!
    '''

    def __init__(self, register, bit):
        self.bit_mask = 1 << (bit % 8)  # the bitmask *within* the byte!
        self.byte = register

    def __get__(self, obj, objtype=None):
        return bool(nvm[self.byte] & self.bit_mask)

    def __set__(self, obj, value):
        if value:
            nvm[self.byte] |= self.bit_mask
        else:
            nvm[self.byte] &= ~self.bit_mask
```